src/uptane/virtualsecondary.cc

```diff
@@ -0,0 +1,35 @@
+#include "uptane/virtualsecondary.h"
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/filesystem.hpp>
+
+#include "crypto.h"
+
+namespace Uptane {
+VirtualSecondary::VirtualSecondary(const SecondaryConfig& sconfig_in) : ManagedSecondary(sconfig_in) {
+  boost::filesystem::create_directories(sconfig.firmware_path.parent_path());
+}
+
+bool VirtualSecondary::storeFirmware(const std::string& target_name, const std::string& content) {
+  Utils::writeFile(sconfig.target_name_path.string(), target_name);
+  Utils::writeFile(sconfig.firmware_path.string(), content);
+  return true;
+}
+
+bool VirtualSecondary::getFirmwareInfo(std::string* target_name, size_t& target_len, std::string* sha256hash) {
+  if (!boost::filesystem::exists(sconfig.target_name_path) || !boost::filesystem::exists(sconfig.firmware_path)) {
+    *target_name = std::string("noimage");
+    target_len = 0;
+    *sha256hash = std::string("");
+    return true;
```

Comment:
maybe return false?

Reply:
No, false is reserved for connection failures, which don't happen in virtual secondaries.